lib/improver/cli/weighted_blending.py

```diff
@@ -173,30 +172,127 @@ def main(argv=None):
     if ((args.wts_calc_method == "nonlinear") and np.any([args.y0val,
                                                           args.ynval])):
         parser.wrong_args_error('y0val, ynval', 'non-linear')
+
     if (args.wts_calc_method == "dict") and not args.wts_dict:
         parser.error('Dictionary is required if --wts_calc_method="dict"')
 
-    # load cubes to be blended
-    cubelist = load_cubelist(args.input_filepaths)
-
     if args.wts_calc_method == "dict":
         with open(args.wts_dict, 'r') as wts:
             weights_dict = json.load(wts)
     else:
         weights_dict = None
 
-    plugin = WeightAndBlend(
-        args.coordinate, args.wts_calc_method,
-        weighting_coord=args.weighting_coord, wts_dict=weights_dict,
-        y0val=args.y0val, ynval=args.ynval, cval=args.cval)
-    result = plugin.process(
-        cubelist, cycletime=args.cycletime,
-        model_id_attr=args.model_id_attr,
-        spatial_weights=args.spatial_weights_from_mask,
-        fuzzy_length=args.fuzzy_length)
+    # Load cubes to be blended.
+    cubelist = load_cubelist(args.input_filepaths)
+
+    result = process(cubelist, weights_dict, args.wts_calc_method,
+                     args.coordinate, args.cycletime, args.y0val, args.ynval,
+                     args.cval, args.weighting_coord, args.model_id_attr,
+                     args.spatial_weights_from_mask, args.fuzzy_length)
 
     save_netcdf(result, args.output_filepath)
 
 
+def process(cubelist, weights_dict, wts_calc_method, coordinate, cycletime,
+            y0val, ynval, cval, weighting_coord,
+            model_id_attr='mosg__model_configuration',
+            spatial_weights_from_mask=False, fuzzy_length=20000.0):
+    """Module to run weighted blending.
+
+    Load in arguments and ensure they are set correctly.
+    Then load in the data to blend and calculate default weights
```

Comment:
calculate weights (not default weights unless some of the arguments are missing).